app/authentication/session_storage.py

```diff
@@ -18,47 +19,56 @@ class SessionStorage:
     def __init__(self, database):
         self._database = database
 
-    def store_user_id(self, user_id):
+    def store_user_id(self, user_id, session_data):
         """
         Create a new eq_session_id and associate it with the user_id specified
         """
         eq_session_id = str(uuid4())
         session[EQ_SESSION_ID] = eq_session_id
-        eq_session = EQSession(eq_session_id, user_id)
+        eq_session = EQSession(eq_session_id, user_id, session_data)
 
         logger.debug('Adding eq_session to database', eq_session_id=eq_session_id, user_id=user_id)
         self._database.add(eq_session)
 
-    def delete_session_from_db(self):
+    def delete_session_from_db(self, eq_session=None):
         """
         deletes user session from database
         """
-        if EQ_SESSION_ID in session:
-            eq_session_id = session[EQ_SESSION_ID]
-            eq_session = self._get_user_session(eq_session_id)
-            if eq_session is not None:
-                self._database.delete(eq_session)
-            else:
-                logger.debug("eq_session_id from user's cookie not found in database")
-        else:
-            logger.debug("eq_session_id does not exist in user's cookie")
+        if eq_session is None:
+            eq_session = self._get_user_session()
+        if eq_session is not None:
+            self._database.delete(eq_session)
 
     def get_user_id(self):
         """
         Retrieves a user's id
         :return: the user's JWT
         """
-        user_id = None
-        if EQ_SESSION_ID in session:
-            eq_session_id = session[EQ_SESSION_ID]
-            eq_session = self._get_user_session(eq_session_id)
-            if eq_session is not None:
-                user_id = eq_session.user_id
+        eq_session = self._get_user_session()
+        if eq_session is not None:
+            return eq_session.user_id
+
+        return None
 
-        return user_id
+    def get_session_data(self):
+        """
+        Retrieves a user's id
+        :return: the user's JWT
+        """
+        eq_session = self._get_user_session()
+        if eq_session is not None:
+            return json.loads(eq_session.session_data)
+
+        return None
 
     @staticmethod
-    def _get_user_session(eq_session_id):
+    def _get_user_session():
```

Comment:
Should we be caching this result?
How many times in a request will this be hit?

Reply:
I did originally cache it however it's only called once per request so not really required. By not caching it will make it easier to use session storage for dynamic values in the future.